seqr/views/utils/dataset_utils.py

```diff
@@ -111,39 +111,41 @@ def _find_or_create_missing_sample_records(
         samples,
         projects,
         user,
-        sample_ids,
+        sample_project_tuples,
+        get_individual_sample_key,
         raise_no_match_error=False,
         raise_unmatched_error_template=None,
         create_active=False,
         get_individual_sample_lookup=_get_individual_sample_lookup,
-        get_individual_sample_key=None,
         **kwargs
 ):
     samples = list(samples)
-    remaining_sample_ids = set(sample_ids) - {sample.sample_id for sample in samples}
+    remaining_sample_keys = set(sample_project_tuples) - {(sample.sample_id, sample.project_name) for sample in samples}
+    remaining_sample_ids = set()
     matched_individual_ids = {sample.individual_id for sample in samples}
-    if len(remaining_sample_ids) > 0:
+    if len(remaining_sample_keys) > 0:
         remaining_individuals_dict = get_individual_sample_lookup(
             Individual.objects.filter(family__project__in=projects).exclude(id__in=matched_individual_ids)
         )
 
         # find Individual records with exactly-matching individual_ids
         sample_id_to_individual_record = {}
-        for sample_id in remaining_sample_ids:
-            individual_key = get_individual_sample_key(sample_id) if get_individual_sample_key else sample_id
+        for sample_key in remaining_sample_keys:
+            individual_key = (get_individual_sample_key(sample_key[0]), sample_key[1])
```

Comment:
Since `get_individual_sample_key` is not optional, we don't have to handle the case when it is None.